src/app/object/components/GotObject.tsx

```diff
@@ -0,0 +1,116 @@
+import { State as RootState } from 'app/redux/state';
+import { fetchType, GotTypeDto, GotTypePropertyDto } from 'app/type';
+import * as React from 'react';
+import { Component } from 'react';
+import { connect } from 'react-redux';
+import { Container, Form, Header, Message } from 'semantic-ui-react';
+import { GotObjectProperty } from './GotObjectProperty';
+
+interface ReduxProps {
+    /**
+     * Action for fetching type descriptions from the redux store.
+     */
+    fetchType: typeof fetchType;
+}
+
+/**
+ * Partial interface allowing all properties to be undefinded for easier
+ * merging.
+ */
+interface PartialProps {
+    /**
+     * Type name unique in the whole got environment to specify the objects type
+     * rendered by this component.
+     */
+    typeName?: string;
+
+    /**
+     * Type declaration which can be empty before the component gets rendered. It will be filled
+     * based on the typeName prop.
+     */
+    type?: GotTypeDto;
+
+    /**
+     * Represents the level in the component tree for rendering purposes.
+     */
+    level?: number;
+
+    /**
+     * Represents an error object in case the component threw an error that should be shown
+     * to the user or generally influence the rendering of the component.
+     */
+    error?: Error;
+}
+
+interface Props extends PartialProps {
+    typeName: string;
```

Comment:
Warum gibt es den einmal als Optional und einmal als fix ?

Reply:
Für die Component ist es ein pflichtfeld, die methode mapStateToProps soll aber nur ein partial objekt zurück geben, in dem nicht alle props gesetzt werden müssen.